### PR TITLE
Fix minor packaging issues in automation lib

### PR DESCRIPTION
**Pull request by jasongin**
This PR fixes some minor issues that affected building and consumption of the automation library externally via an npm package:
  - Remove the `postinstall` npm script. This should not be run after installing the automation package because the npm package includes only the already-compiled output. And that script is not necessary for vscode smoketests because the `smoke` package runs `yarn --cwd ../automation compile` during its compilation.
  - Fix the `driver` module export in `index.ts`.
  - Add `.npmignore`
  - Update `.gitignore`

None of these changes should have any impact on building or running the vscode smoke tests. Anyway, I verified that the vscode smoke tests are still passing.

### CODE REVIEW

**Comment by jasongin** [comment]
@Tyriar

**Review by Tyriar — CHANGES_REQUESTED** [review]

File: test/automation/package.json [hunk 1]
```diff
@@ -9,7 +9,6 @@
   "main": "./out/index.js",
   "private": true,
   "scripts": {
-    "postinstall": "npm run compile",
```
[the remark this answers is not in the document]
Comment (joaomoreno) [reply test/automation/package.json]: Interesting... for some reason that didn't work for me when I last edited the smoke test, I really had to start two watch tasks. :thinking:

[the remark this answers is not in the document]
Comment (Tyriar) [reply test/automation/package.json]: This is fine to merge if it's `prepare`, sounds like we can also change the README to only watch in test/smoke?

[the remark this answers is not in the document]
Comment (jasongin) [reply test/automation/package.json]: I added the `prepare` script and removed the extra step from the README. But maybe one of you should verify `yarn watch` from the smoke directory is working as you expect. It seems for me...

**Review by Tyriar — APPROVED** [review]
Works fine here 👍

**Comment by Tyriar** [comment]
Hmm, using && didn't used to work on Windows but it seems fine now in both Pipelines and Actions, guess that's fine.

**Comment by jasongin** [comment]
Thanks!

**Pull request merged**